api/client.go

```diff
@@ -670,24 +709,41 @@ func (c *Client) SetBackoff(backoff retryablehttp.Backoff) {
 // Also, only the client's config is currently copied; this means items not in
 // the api.Config struct, such as policy override and wrapping function
 // behavior, must currently then be set as desired on the new client.
+//
+// One exception to this is the addr field of Client, which is also copied over.
+// Otherwise you have the confusing result of creating a client, customizing
+// the address via client.SetAddress() then cloning it and having it show the
+// default address again, instead of the one you just set. I'm not even sure
+// why the client has addr as a member at all, other than the convenience of
+// having it pre-parsed into a URL instead of a string.
 func (c *Client) Clone() (*Client, error) {
 	c.modifyLock.RLock()
```

Comment:
Why not just grab both locks at the outset and defer release them?  They're just read locks, and this way the code will be easier to read and reason about.

Reply:
Sure